backend/fms_core/migrations/0017_v3_1_0.py

```diff
@@ -12,16 +12,34 @@ def add_values_to_sample_volume(apps, schema_editor):
         sample.volume = float_to_decimal(sample.volume_history[-1]["volume_value"] if sample.volume_history else 0.0)
         sample.save()
 
+
+def handle_sample_versions(apps, schema_editor):
```

Comment:
Nice idea to regroup them in a single iteration ... We only need to be aware that they are no longer in the same transaction (since they are in a different migration) as the rest of the related operations.